src/main/java/com/exasol/adapter/document/documentfetcher/files/randomaccessinputstream/InMemoryRandomAccessStream.java

```diff
@@ -0,0 +1,62 @@
+package com.exasol.adapter.document.documentfetcher.files.randomaccessinputstream;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import com.exasol.errorreporting.ExaError;
+
+/**
+ * {@link RandomAccessInputStream} that simply loads all contents of an input stream to an byte array and by that offers
+ * random access.
+ */
+public class InMemoryRandomAccessStream extends RandomAccessInputStream {
+    private final byte[] data;
+    long position = 0;
+
+    public InMemoryRandomAccessStream(final byte[] data) {
+        this.data = data;
+    }
+
+    /**
+     * Create an {@link InMemoryRandomAccessStream} with the content from a regular {@link InputStream}.
+     *
+     * @param normalInputStream input stream to read from
+     * @return built {@link InMemoryRandomAccessStream}
+     */
+    public static InMemoryRandomAccessStream getInMemoryCache(final InputStream normalInputStream) {
+        try {
+            return new InMemoryRandomAccessStream(normalInputStream.readAllBytes());
+        } catch (final IOException exception) {
+            throw new IllegalStateException(ExaError.messageBuilder("E-VSDF-12").message(
+                    "This data source does not support random access for files. So we tried to keep everything in memory, but it failed (probably due to too few memory).")
```

Comment:
We should write this to user guide that it is good practice to have < 1GB parquet files. UDFs also have limited memory.

Reply:
Yes, but only for data sources that don't support random access read. This class is only used as a polyfill for those data sources (Currently none; only for testing).